lib/datadog/tracing/contrib/propagation/sql_comment.rb

```diff
@@ -0,0 +1,49 @@
+# typed: false
+
+require_relative 'sql_comment/comment'
+require_relative 'sql_comment/ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Propagation
+        # Implements sql comment propagation related contracts.
+        module SqlComment
+          def self.annotate!(span_op, mode)
+            return unless mode.enabled?
+
+            span_op.set_tag(Ext::TAG_DBM_TRACE_INJECTED, true) if mode.full?
+          end
+
+          def self.prepend_comment(sql, span_op, mode, tags: {})
+            return sql unless mode.enabled?
+
+            tags.merge!(service_context)
```

Comment:
`service_context` is always overriding the values passed in `tags`. This means it is not possible, with this implementation, to override the service name.

I actually recommend reducing the API a bit here, and only accepting `service_name: nil`, instead of `tags: {}`. I don't see a scenario where we want to tag with a custom `env` or custom `version`, I only see `service` as mutable here. And we should make sure if service is provided, we don't override it with `datadog_configuration.env`.